packages/@o3r/rules-engine/src/services/rules-engine.effect.ts

```diff
@@ -61,35 +62,52 @@ export class PlaceholderTemplateResponseEffect {
     )
   );
 
-  constructor(private actions$: Actions, private rulesEngineService: RulesEngineService, private dynamicContentService: DynamicContentService) {
+  constructor(private actions$: Actions, private rulesEngineService: RulesEngineService, private dynamicContentService: DynamicContentService, private translationService: LocalizationService) {
   }
 
   /**
-   * Renders the html template, replacing facts and urls
+   * Renders the html template, replacing facts and urls and localizationKeys
    *
    * @param template
    * @param vars
    * @param facts
    */
-  public async renderHTML(template?: string, vars?: Record<string, PlaceholderVariable>, facts?: { name: string; factValue: any }[]) {
```

Comment:
this is actually a breaking change, maybe you can reprovide the `renderHTML` as `depdecated` with a `latestValueFrom(this.getRenderedHTML$)`

Reply:
That is what I first intended - hence the new name - but we said it was internal logic and was ok to clean up. I can leave it for safety reasons, it does not matter much anyway

Reply:
I'm not sure it made sense to be public to be honest.
Maybe the new one should be private